### PR TITLE
Add the last set of top-level sigs, and enable GHC warnings

### PR DESCRIPTION
**Pull request by bpfoley**
While here, turn on GHC tab warnings, and remove a duplicate 'head -1'
in the Makefile.

### CODE REVIEW

File: src/comp/BackendNamingConventions.hs [hunk 1]
```diff
@@ -116,12 +122,13 @@ isCRegInst avi = (isCRegN avi) || (isCRegUN avi) || (isCRegA avi)
 -- ---------------
 -- Names of CReg methods
 -- (inlining happens before methods are renamed to ports)
-
```
Comment (quark17) [inline src/comp/BackendNamingConventions.hs]: Don't delete this blank line; the comment is for the larger group of functions not just these first three

File: src/comp/BackendNamingConventions.hs [hunk 2]
```diff
@@ -132,25 +139,29 @@ cregWriteArgId i n = mkMethId i (cregWriteId n) Nothing (MethodArg 1)
 
 -- ---------------
 -- Names of ports and parameters on primtive CReg
-
```
Comment (quark17) [inline src/comp/BackendNamingConventions.hs]: I would also not delete this blank

File: src/comp/Makefile [hunk 3]
```diff
@@ -84,7 +84,7 @@ WISHFLAGS += $(EXTRAWISHLIBS)
 GHC ?= ghc
 GHCJOBS ?= 1
 
-GHCVERSION=$(shell $(GHC) --version | head -1 | egrep -o "[0-9]+\.[0-9]+\.[0-9]" | head -1)
+GHCVERSION=$(shell $(GHC) --version | head -1 | egrep -o "[0-9]+\.[0-9]+\.[0-9]" )
```
Comment (quark17) [inline src/comp/Makefile]: I don't know enough about this command (and the range of possible GHC version strings) to know if this is safe to do?

Comment (bpfoley) [reply src/comp/Makefile]: It doesn't really matter: `ghc --version | head -1` is only ever going to print one line, and the `egrep -o` prints out a subset of that line, so the final `| head -1` is definitely redundant.

**Pull request merged**